pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,164 @@
+package ingress
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestGatewayApiControllerMode(t *testing.T) {
+	NewTest(t).Id("T41").Groups(Full, InterOp).Run(func(t TestHelper) {
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_3) {
+			t.Skip("TestGatewayApiControllerMode was added in v2.3")
+		}
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		ossm.DeployControlPlane(t)
+
+		smcpName := "basic"
+
+		t.LogStep("Install Gateway API CRD's")
+		shell.Executef(t, "kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null ||   { kubectl kustomize github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.5.1 | kubectl apply -f -; }")
```

Comment:
```suggestion
		shell.Executef(t, "kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null && echo 'Gateway API CRDs already installed' || kubectl apply -k github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.5.1")
```

Also, any specific reason for using version v0.5.1 instead of v0.6.2, for example?

Reply:
Istio 1.16 is configured to work with 0.5.1, https://istio.io/v1.16/docs/setup/additional-setup/getting-started/